src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -58,9 +58,13 @@ public LitematicaSchematic(CompoundTag nbt) {
      * @return Array of subregion tags.
      */
     private static CompoundTag[] getRegions(CompoundTag nbt) {
-        return nbt.getCompound("Regions").getAllKeys().stream()
-                .map(nbt.getCompound("Regions")::getCompound)
-                .toArray(CompoundTag[]::new);
+        return nbt.getCompound("Regions")
+            .map(CompoundTag::values)
+            .map(r -> r.stream()
+                .filter(v -> v instanceof CompoundTag)
+                .map(CompoundTag.class::cast)
+                .toArray(CompoundTag[]::new)
+            ).orElse(new CompoundTag[0]);
```

Comment:
This looks like it also fixes an NPE with malformed schematics. Should this be backported?

Reply:
up to you ig

its not necessarily incorrect for malformed inputs to throw exceptions

this optional chaining is more natural to do now that the nbt api is returning them

but i think there is existing code up the stack that would catch and log those exceptions anyway